app/models/user.rb

```diff
@@ -14,6 +14,34 @@ class User < ApplicationRecord
   validates :phone_number, aus_phone_number: true
   validates :unconfirmed_phone_number, aus_phone_number: true
 
+  after_save :after_save
+
+  def after_save
```

Comment:
I'd recommend extracting all of this code out into a service object. It makes it easier to test and read and its reusable.

    def after_save
      RecordChangeLogger.new(self).perform
    end

    class RecordChangeLogger
      REDACT_KEYS = %w(encrypted_password ...)

      def initialize(record)
        @record = record
        @changes = determine_changes
      end

      def perform
        if @changes['encrypted_password']

        # etc
      end
 
      private
      def determine_changes
        self.changes.inject({}) do |h, (k, v)|
          if REDACT_KEYS.include?(k)
            h[k] =['REDACTED', 'REDACTED']
          else
            h[k] =v
          end
          h
        end
      end
    end